### PR TITLE
Add benchmarks to pallet-gear

### PR DESCRIPTION
**Pull request by ekovalev**

### CODE REVIEW

File: runtime/Cargo.toml [hunk 1]
```diff
@@ -58,7 +58,7 @@ pallet-gear = { version = "2.0.0", default-features = false, path = "../pallets/
 gear-rpc-runtime-api = { version = "2.0.0", default-features = false, path = "../pallets/gear/rpc/runtime-api" }
 
 [build-dependencies]
-substrate-wasm-builder = { version = "4.0.0", git = "https://github.com/paritytech/substrate.git" }
+substrate-wasm-builder = { version = "4.0.0" }
```
Comment (shamilsan) [inline runtime/Cargo.toml:61]:
```suggestion
substrate-wasm-builder = "4.0.0"
```

File: pallets/gear/src/benchmarking.rs [hunk 2]
```diff
@@ -16,27 +16,213 @@
...
 
-//! Benchmarking setup for pallet-template
+//! Gear pallet benchmarking
+
+#![cfg(feature = "runtime-benchmarks")]
 
 use super::*;
+use codec::Encode;
+use common::{IntermediateMessage, Origin};
+use parity_wasm::elements::{
```
Comment (shamilsan) [inline pallets/gear/src/benchmarking.rs:26]: Philosophic question :-)
Isn't better to use `use parity_wasm::elements::*;` when we import so many subjects?

Comment (ekovalev) [reply pallets/gear/src/benchmarking.rs:26]: Agree, it'd be more readable that way.

**Comment by NikVolf** [comment]
I thought benchmarking should auto-generate weight formulas which we will attach to our runtime calls?

**Review by NikVolf — APPROVED** [review]

**Comment by ekovalev** [comment]
> I thought benchmarking should auto-generate weight formulas which we will attach to our runtime calls?

They do: you need to run `gear-node benchmark` CLI command with `--output` flag. I thought of including default weights for extrinsics based on benchmarking in a next PR because it's essentially a different task.

**Pull request merged**